docs/edgeql/expressions/parameters.rst

```diff
@@ -0,0 +1,171 @@
+.. _ref_eql_expr_params:
+
+Parameters
+==========
+
+:edb-alt-title: Query Parameters
+
+A parameter reference is used to indicate a value that is supplied externally
+to an EdgeQL expression.
+
+An example of query, that can be parametrized:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT Person { birth_date } FILTER .first_name = 'John';
+    {
+        Object {birth_date: <cal::local_date>'1971-01-01'},
+    }
+
+The equivalent parametrized query:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT Person { birth_date } FILTER .first_name = <str>$name;
+    Parameter <str>$name: John
+    {
+        Object {birth_date: <cal::local_date>'1971-01-01'},
+    }
+
+Note how the actual parameter value is asked by REPL in a separate prompt,
+after the whole query is accepted.
+
+Here's how parameters can be used with language bindings. e.g. with Python:
+
+.. code-block:: python
+
+    await connection.fetchone(
+        "SELECT Person { birth_date } FILTER .first_name = <str>$name",
+        name="John")
+
+And with JavaScript:
+
+.. code-block:: javascript
+
+    await connection.fetchOne(
+        "SELECT Person { birth_date } FILTER .first_name = <str>$name",
+        {name: "John"})
+
+
+Parameter Types
+---------------
+
+The following values can be turned into parameters:
+
+* Any :ref:`scalar type <ref_datamodel_scalar_types>`
+* :eql:type:`Array <array>` of any scalar type
+
+The parameter type must be specified in angular brackets in front of the variable
+name (the syntax is similar to type casts):
+
+.. code-block:: edgeql
+
+    INSERT Person {
+        first_name := <str>$first_name,
+        age := <int32>$age,
+        interests := <array<str>>$interests,
+    }
+
+See :ref:`ref_eql_types` for more info on specifying types.
+
+
+Optional Parameters
+-------------------
+
+By default, query parameters are ``REQUIRED`` which means that the query would fail
+if parameter value is nil. You can use ``OPTIONAL`` keyword for optional
```

Comment:
Then this:

```suggestion
By default, query parameters are ``REQUIRED`` which means that the query would fail
if parameter value is an empty set. You can use ``OPTIONAL`` keyword for optional
```

Reply:
I'm sure that "empty set" would be confusing to me. `set()` is an empty set in python. But, let's wait for user feedback.